### DISCUSSION TITLE
Cylance blocking freeplane versions after / newer than 1.11.7. Any possibly relevant change since 1.11.8?

### DISCUSSION THREAD

**Discussion by chipbite** [freeplane/freeplane#1890] [Help]
**Cylance Protect blocks Freeplane.exe versions 1.11.8 and later**
**_Please note:_** This is an issue but whether it can be deemed a bug in the freeplane code is debatable and undetermined. This is an issue and the exact cause / actual problem is not clear.

https://en.wikipedia.org/wiki/Cylance
https://www.blackberry.com/us/en/products/blackberry-cybersecurity

**To Reproduce**
1. Ensure that you are on a corporate machine with Cylance Protect running. I am running windows, have not tested other OS with Cylance (e g macos, I could test it later).
2. Download and install any version of Freeplane up to and including v. **1.11.7** - run and it works without issue.
3. Download and install Freeplane, any version from **1.11.8** and later. 
4. Attempt to run. 

**Expected behavior**
Versions later than 1.11.7 (i e starting from 1.11.8) should also run?

**Actual behavior**
All* versions from 1.11.8 and later are blocked by Cylance Protect.

(* full disclosure, I have not tested all - these are the three versions I have actually tried and tested: 1.11.8, 1.11.11 and 1.11.14).

**Screenshots**
<img width="305" alt="2024-06-24 15_07_59-Freeplane - Wikipedia" src="https://github.com/freeplane/freeplane/assets/559199/95902159-aeab-4bbf-a55c-1c67b9f7b5c9">

<img width="300" alt="image" src="https://github.com/freeplane/freeplane/assets/559199/5fe25b70-d9d1-4f67-8664-ff7dc5263bb4">

**Desktop (please complete the following information):**
 - OS: **Microsoft Windows [Version 10.0.19045.4412]**
 - Version [e.g. 22]: (see above)
 - Java Version **openjdk 21.0.3 2024-04-16 LTS**

**Additional context**
Since later versions work, I am ofc not sure what the issue might be. I have reported the issue to our internal IT support. Hoping they might bring it up with Cylance, but not sure if that will happen and if Cylance would care. 

# Why report here? Q: Has anything changed starting with version 1.11.8 that might cause this?
The main reason for reporting this here is that earlier versions work. The reason might ofc be that earlier versions are whitelisted with Cylance, and later versions (1.11.8 and on) are not. The reason may also be a change in freeplane with this version. So, I wonder if there might be some change in freeplane that may cause Cylance to react? Generation, rewriting - anything really...

If I am getting this issue, other corporate citizens with Cylance installed will also likely run into this, and they may ofc not be as perseverant as I am... :-)  

A long shot, but I figured I must check.

Thanks for a great piece of software and for reading my issue!

**Comment by dpolivaev** [comment]
Since Freeplane 1.11.9 I use a modified version of the launcher, you can find the modifications in https://github.com/dpolivaev/launch4j

**Comment by chipbite** [comment]
Thanks, yes, it seems not unlikely to be the culprit: Seems to be several issues of it getting flagged by security / antivirus software. 

https://sourceforge.net/p/launch4j/bugs/search/?q=Virus

<img width="932" alt="image" src="https://github.com/freeplane/freeplane/assets/559199/09a2161f-896f-4b82-abbf-3b2854ab6cab">



Ofc, it's false positives, but still an issue for users. 

Thanks again for your reply.

**Comment by chipbite** [comment]
Hi again! 
Ok, I have also filed a report with launch4j.
https://sourceforge.net/p/launch4j/bugs/234/

> Since Freeplane 1.11.9 I use a modified version of the launcher, you can find the modifications in https://github.com/dpolivaev/launch4j

I dont think I have any input to this, I am an end-user with rather obsolete and limited java skills.

I worry that `1.11.7` will be the last update I can use. 

### Is there any workaround that would be possible?

A different installer or running the jar or java classes "directly"? 
(I am winging ideas that I am not sure if they are applicable. Again, my java knowledge is fairly limited and non-current)

**Comment by macmarrum** [comment]
Have you tried launching via freeplane.bat? It's installed into the same folder as freeplane.exe

**Reply by chipbite** [reply]
No, I have not. Thanks for that, @macmarrum. I was thinking along those lines, but had not checked. 
I will install a new version again and try it.

Thanks!
